pkg/gatherers/clusterconfig/node_logs.go

```diff
@@ -0,0 +1,147 @@
+	clientSet, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+	return gatherNodeLogs(ctx, clientSet.CoreV1())
+}
+
+func gatherNodeLogs(ctx context.Context, client corev1client.CoreV1Interface) ([]record.Record, []error) {
+	nodes, err := client.Nodes().List(ctx, metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master"})
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	records, errs := nodeLogRecords(ctx, client.RESTClient(), nodes)
+	return records, errs
+}
+
+// nodeLogRecords generate the records and errors list
+func nodeLogRecords(ctx context.Context, restClient rest.Interface, nodes *corev1.NodeList) ([]record.Record, []error) {
+	var errs []error
+	records := make([]record.Record, 0)
+
+	bufferSize := recorder.MaxArchiveSize * logCompressionRatio / len(nodes.Items) / 2
+	klog.V(2).Infof("Maximum buffer size: %v bytes", bufferSize)
+	buf := bytes.NewBuffer(make([]byte, 0, bufferSize))
+
+	for i := range nodes.Items {
+		name := nodes.Items[i].Name
+		uri := nodeLogResourceURI(restClient, name)
+		req := requestNodeLog(restClient, uri, logNodeMaxTailLines, logNodeUnit)
+
+		logString, err := nodeLogString(ctx, req, buf, bufferSize)
+		if err != nil {
+			klog.V(2).Infof("Error: %q", err)
+			errs = append(errs, err)
+		}
+
+		records = append(records, record.Record{
+			Name: fmt.Sprintf("config/node/logs/%s.log", name),
+			Item: marshal.Raw{Str: logString},
+		})
+	}
+
+	return records, errs
+}
+
+// nodeLogResourceURI creates the resource path URI to be fetched
+func nodeLogResourceURI(client rest.Interface, name string) string {
+	return client.Get().
+		Name(name).
+		Resource("nodes").SubResource("proxy", "logs").
+		Suffix("journal").URL().Path
+}
+
+// requestNodeLog creates the request to the API to retrieve the resource stream
+func requestNodeLog(client rest.Interface, uri string, tail int, unit string) *rest.Request {
+	return client.Get().RequestURI(uri).
+		SetHeader("Accept", "text/plain, */*").
+		SetHeader("Accept-Encoding", "gzip").
+		Param("tail", strconv.Itoa(tail)).
+		Param("unit", unit)
+}
+
+// nodeLogString retrieve the data from the stream, decompress it (if necessary) and return the string
+func nodeLogString(ctx context.Context, req *rest.Request, out *bytes.Buffer, size int) (string, error) {
+	in, err := req.Stream(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer in.Close()
+
+	buf := bufio.NewReaderSize(in, size)
+	head, err := buf.Peek(1024)
+	if err != nil && err != io.EOF {
+		return "", err
+	}
+
+	_, err = gzip.NewReader(bytes.NewBuffer(head))
+	if err != nil {
+		// not gzipped stream
+		_, err = io.Copy(out, buf)
+		if err != nil && err != io.EOF {
+			return "", err
+		}
+	} else {
+		r, err := gzip.NewReader(buf)
+		if err != nil {
+			return "", err
+		}
+
+		// nolint: gosec
+		_, err = io.Copy(out, r)
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
Sorry but I still don't understand why all this is needed. I think you can do just something like:
```
	r, err := gzip.NewReader(in)
	if err != nil {
		scanner = bufio.NewScanner(in)
	} else {
		scanner = bufio.NewScanner(r)
	}
```
or not?

Reply:
It doesn't work, u need to "uncompress" the streaming first. Otherwise, it will crash. This logic is basically the same that is being used for the `oc` command. I trust them.

Reply:
Well it seems to work fine in my cluster...

Reply:
OK, so, maybe I can simplify it. Let me give it a try.

Reply:
Nope. I tested doesn't crash, but what I get are empty files.

Reply:
Can you please share your code? It worked for me and the files were not empty

Reply:
Fixed.

Reply:
Thank you.